src/components/my-widgets-page.jsx

```diff
@@ -75,30 +80,40 @@ const MyWidgetsPage = () => {
 		if (state.selectedInst && permUsers) {
 			const isEditable = state.selectedInst.widget.is_editable === "1"
 			const othersPerms = new Map()
-			for(const i in permUsers.widget_user_perms){
+			for (const i in permUsers.widget_user_perms) {
 				othersPerms.set(i, rawPermsToObj(permUsers.widget_user_perms[i], isEditable))
 			}
 			let _myPerms
-			for(const i in permUsers.user_perms){
+			for (const i in permUsers.user_perms) {
 				_myPerms = rawPermsToObj(permUsers.user_perms[i], isEditable)
 			}
-			setState({...state, otherUserPerms: othersPerms, myPerms: _myPerms})
+			setState({ ...state, otherUserPerms: othersPerms, myPerms: _myPerms })
 		}
 	}, [state.selectedInst, JSON.stringify(permUsers)])
 
 	// isLoading - initial data, from cache
 	// isFetching - actual data, from API
 	useEffect(() => {
 		if (!isFetching) {
-			checkPreselectedWidgetAccess(widgets)
+			if (page <= data.total_num_pages) {
+				setPage(page + 1)
+			}
+			setWidgetsList(current => [...current, ...data.pagination])
 		}
 	}, [isFetching])
```

Comment:
Attaching a `useEffect` to `isFetching` is a little unconventional, but since that useEffect was already being used, we may as well roll with it. In addition to my other suggestion for disabling the query once all pages are fetched, I think it might be useful to assign `isFetching` to a value with a name that makes it easier to identify. You can see examples of this on lines 59 and 64, something like:
```
const { data: instancesFetching } = useQuery(...)
```

Obviously the references to `isFetching` on lines 97 and 103 would have to be updated to the new value.